### DISCUSSION TITLE
When is 4.0 going to release?

### DISCUSSION THREAD

**Discussion by ryan6416** [SelfControlApp/selfcontrol#693] [Q&A] [answered]
Excited for the M1 native support! @cstigler

**Comment by karbassi** [comment]
@cstigler recently bumped the version to 4.0 final in https://github.com/SelfControlApp/selfcontrol/commit/86b540e567aae9847a8d83d0744394532e2b19b7

I'd love to see a release soon.

**Comment by cstigler** [comment]
Glad y'all are excited! I know it's been a wait. This has been on my todo list for multiple months - life has been busy 😅

All that's left is the final testing and release process. I'd love to say it'll happen this weekend (I'm hoping to make some time), but I hate to promise dates when I'm not sure I'll hit them. So: soon!

**Reply by karbassi** [reply]
I'm trying to help by making a github action that will build and test for you.

**Reply by cstigler** [reply]
Aw that's awesome, thank you! I think that'd be a really good start to more automation. Unfortunately, we're also lacking much automated testing, so the testing process is 95% manual right now. Automating the testing process would be incredible, but is also a massive effort, because the tests involve complex flows/situations like multi-user, browser testing, etc...

**Reply by karbassi** [reply]
@cstigler it's not fully working, but here's a start. https://github.com/SelfControlApp/selfcontrol/pull/695

**Comment by cstigler** [comment, marked as answer]
4.0 is out :) download at https://selfcontrolapp.com/

I'll push the appcast update (so it auto-updates on everyone's systems) sometime this weekend, after making sure it's not borking things for everyone

**Reply by karbassi** [reply]
@cstigler you can mark this as 'answered' and close it.

**Reply by cstigler** [reply]
Good call! I can mark it answered but I don't see a "close" option - only lock and delete, neither of which is really what I want. I guess I'll just leave it?
